src/extension/commands_helpers.c

```diff
@@ -383,14 +428,19 @@ dd_result dd_command_proc_resp_verd_span_data(
             verd_str);
     }
 
-    bool should_block = dd_mpack_node_lstr_eq(verdict, "block");
+    dd_result res = dd_success;
     // Parse parameters
-    if (should_block) {
+    if (dd_mpack_node_lstr_eq(verdict, "block")) {
+        res = dd_should_block;
         _command_process_block_parameters(mpack_node_array_at(root, 1));
         dd_tags_add_blocked();
+    } else if (dd_mpack_node_lstr_eq(verdict, "redirect")) {
+        res = dd_should_redirect;
+        _command_process_redirect_parameters(mpack_node_array_at(root, 1));
+        dd_tags_add_blocked();
     }
 
-    if (should_block || dd_mpack_node_lstr_eq(verdict, "record")) {
+    if (res == dd_should_block || dd_mpack_node_lstr_eq(verdict, "record")) {
```

Comment:
We should still record an event on redirect.

Reply:
Done